lib/cog/models/group.ex

```diff
@@ -28,12 +29,30 @@ defmodule Cog.Models.Group do
   detail_fields [:id, :name]
 
   @doc """
-  Creates a changeset based on the `model` and `params`.
-
   If no params are provided, an invalid changeset is returned
   with no validation performed.
   """
-  def changeset(model, params \\ :empty) do
+  def changeset(model), do: changeset(model, :empty)
+
+  @doc """
+  Creates a changeset based on the `model` to validate a delete
+  action.
+  """
+  def changeset(model, :delete) do
+    changeset = %{Changeset.change(model) | action: :delete}
+
+    if model.name == Cog.admin_group do
+      changeset
+      |> add_error(:name, "admin group may not be modified")
```

Comment:
Might want to change the error message to `"admin group may not be deleted"`.

Of course, since we're checking against a specific name, we probably also need to prevent the name from being changed, as well.